### DISCUSSION TITLE
Writing USD with references

### DISCUSSION THREAD

**Discussion by cking-spire** [Autodesk/maya-usd#2192] [Q&A]
If I have imported a USD stage into Maya as regular Maya objects but with referencing turned on, is there a way to write this back to USD keeping those references?

I am building tools to allow artists to assemble sets from USD components. I know I can do this by creating a stage in the USD Layer Editor and adding USD references. I'm just wondering if it is possible to give artists the option of working with regular Maya objects until it comes time to export.

Thanks,
Chris King
Spire Animation Studios

**Comment by womanyoyoyo-adsk** [comment]
Hi Chris,

Currently there is no way to export Maya data to USD with USD referencing enabled. If your Maya data has Maya references, when exporting to USD the exporter will flatten the Maya references to create one USD file.

Also, if you're importing a USD file with USD references, our importer will flatten those USD references into Maya objects (it does not convert the USD references into Maya references).  

Thanks for explaining the workflow you'd like to see. We are working on a workflow that includes a Maya Reference schema that I think would fit your workflow. More details to come.

**Comment by dalorin** [comment]
Hi there

I'm attempting a similar workflow but with USD Proxy shapes instead of Maya references. Does the same apply?

The arnold USD exporter handles the proxy nodes but unfortunately translates them into ArnoldUsd primitives instead of plain old USD references.

Regards

**Comment by frankchen211** [comment]
`pxrUsdReferenceAssembly` node and `pxrUsdExport` can do that but unfortunately they are deprecated, but this is really helpful to build a set/layout workflow.
